tests/testthat/setup-workflow-ref.R

```diff
@@ -194,6 +194,15 @@ create_rlg_models <- function() {
   )
 }
 
+clean_test_enviroment <- function(.f, env = parent.frame())
+{
+  cleanup()
```

Comment:
@kyleam I tried to address this in this commit. Is this what you meant? [https://github.com/metrumresearchgroup/bbr/pull/484/commits/c066cb13e7625d17a28e8e9bdd7ebeea41062742](url)

Reply:
It includes some of the pieces, but doesn't wire things up.

Here's the current situation (again, this is just based on looking at the code; it's worth confirming whether I'm wrong):

```R
f1 <- function(env = parent.frame()) {  # => cleanup
  message(":: f1")
  x_in_f1 <- NULL

  rlang::inform(c("environment:", ls(env)))
}

f2 <- function(env = parent.frame()) {  # => clean_test_enviroment
  message(":: f2")
  x_in_f2 <- NULL

  f1()
  withr::defer(f1(), envir = env)
}

f3 <- function() {  # => calling test function
  message(":: f3")
  x_in_f3 <- NULL

  f2()
}

f3()
```

output:

```
:: f3
:: f2
:: f1
environment:
• env
• x_in_f2
:: f1
environment:
• env
• x_in_f2
```

Notice that `ls()` in this example (and `rm()` in the code I'm commenting on) is working on the wrong environment: the parent rather than the grandparent.

Here's my suggested fix:

```diff
diff --git a/scratch.R b/scratch-fixed.R
index 87aab12..1b97953 100644
--- a/scratch.R
+++ b/scratch-fixed.R
@@ -9,8 +9,8 @@
   message(":: f2")
   x_in_f2 <- NULL

-  f1()
-  withr::defer(f1(), envir = env)
+  f1(env)
+  withr::defer(f1(env))
 }

 f3 <- function() {  # => calling test function
```

```
:: f3
:: f2
:: f1
environment:
• x_in_f3
:: f1
environment:
• x_in_f3
```

So, concretely, that'd be something like this:

```diff
diff --git a/tests/testthat/setup-workflow-ref.R b/tests/testthat/setup-workflow-ref.R
index bce26fb..dc8d246 100644
--- a/tests/testthat/setup-workflow-ref.R
+++ b/tests/testthat/setup-workflow-ref.R
@@ -196,7 +196,7 @@ create_rlg_models <- function() {
 
 clean_test_enviroment <- function(.f, env = parent.frame())
 {
-  cleanup()
+  cleanup(env)
   .f()
   withr::defer(cleanup(), envir = env)
 }
@@ -217,11 +217,11 @@ cleanup <- function(env = parent.frame()) {
   if (fs::dir_exists(LEVEL2_DIR)) fs::dir_delete(LEVEL2_DIR)
 
   # delete model objects from memory
-  suppressSpecificWarning(rm(mod1, pos = parent.frame()), .regexpr = "object.+not found")
-  suppressSpecificWarning(rm(mod2, pos = parent.frame()), .regexpr = "object.+not found")
-  suppressSpecificWarning(rm(mod3, pos = parent.frame()), .regexpr = "object.+not found")
-  suppressSpecificWarning(rm(mod4, pos = parent.frame()), .regexpr = "object.+not found")
-  suppressSpecificWarning(rm(log_df, pos = parent.frame()),.regexpr = "object.+not found")
+  suppressSpecificWarning(rm(mod1, pos = env), .regexpr = "object.+not found")
+  suppressSpecificWarning(rm(mod2, pos = env), .regexpr = "object.+not found")
+  suppressSpecificWarning(rm(mod3, pos = env), .regexpr = "object.+not found")
+  suppressSpecificWarning(rm(mod4, pos = env), .regexpr = "object.+not found")
+  suppressSpecificWarning(rm(log_df, pos = env), .regexpr = "object.+not found")
 }
 
 #' Temporarily perturb a file
```


But again, please don't take my word that there really is an issue in the first place.

Reply:
Thanks for looking into this. I will make sure to play around with it a bit.